### PR TITLE
feat: github workflow for building npm on tag push

### PR DESCRIPTION
**Pull request by dmitry-yudakov**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          elif [[ $GITHUB_REF == refs/tags/v*-* ]]; then
+            NPM_TAG=next
+          elif [[ $GITHUB_REF == refs/tags/v* ]]; then
+            NPM_TAG=latest
+          fi
+          echo "NPM_TAG=${NPM_TAG}" >> $GITHUB_ENV
+          echo "npm_tag=${NPM_TAG}" >> $GITHUB_OUTPUT
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '16'
```
Comment (jor-rit) [inline .github/workflows/publish.yml]: 16? sure about that? I though client already required 20? :)

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        name: Install dependencies
+        run: npm ci --ignore-scripts
+      - id: build
+        name: NPM build
+        run: npm run build
+      - id: publish
+        name: Publish NPM package
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: npm publish --tag "${{ steps.npm-package-tag.outputs.npm_tag }}"
```
Comment (jor-rit) [inline .github/workflows/publish.yml:53]: which repo? there is no regisrty configured in the package.json. So this would try to upload to npm.com

**Review by jor-rit — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        run: npm ci --ignore-scripts
+      - id: build
+        name: NPM build
+        run: npm run build
+      - id: publish
+        name: Publish NPM package
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: npm publish --tag "${{ steps.npm-package-tag.outputs.npm_tag }}"
```
Comment (jor-rit) [inline .github/workflows/publish.yml:53]: the step (id) is called `package-tag` above, while `steps.npm-package-tag.outputs` is used here

**Review by jor-rit — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 3 above)
Comment (dmitry-yudakov) [reply .github/workflows/publish.yml:53]: huge thx! works now :)

**Pull request merged**